web-sender/src/test/java/org/openmrs/eip/web/sender/SenderSyncArchiveControllerTest.java

```diff
@@ -0,0 +1,77 @@
+package org.openmrs.eip.web.sender;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.openmrs.eip.app.SyncConstants.MGT_DATASOURCE_NAME;
+import static org.openmrs.eip.app.SyncConstants.MGT_TX_MGR;
+
+import java.util.List;
+import java.util.Map;
+
+import org.junit.Test;
+import org.openmrs.eip.app.management.entity.sender.SenderSyncArchive;
+import org.openmrs.eip.app.sender.BaseSenderTest;
+import org.openmrs.eip.component.Constants;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.context.jdbc.SqlConfig;
+
+@Sql(scripts = "classpath:mgt_sender_sync_archive.sql", config = @SqlConfig(dataSource = MGT_DATASOURCE_NAME, transactionManager = MGT_TX_MGR))
+public class SenderSyncArchiveControllerTest extends BaseSenderTest {
+	
+	@Autowired
+	private SenderSyncArchiveController controller;
+	
+	@Test
+	public void shouldGetAllArchivesMessages() {
+		Map result = controller.getAll();
+		assertEquals(2, result.size());
+		assertEquals(3, result.get("count"));
+		assertEquals(3, ((List) result.get("items")).size());
+		
+	}
+	
+	@Test
+	public void shouldDosearchByEventDate() {
+		
+		String stardDate = "2022-10-25";
+		String endDate = "2022-10-30";
+		
+		Map results = controller.doSearchByPeriod(stardDate, endDate, SenderSyncArchive.EVENT_DATE);
+		assertEquals(1, results.get("count"));
+		assertEquals(2, results.size());
+		assertNotNull(((List<SenderSyncArchive>) results.get("items")).get(0).getDateCreated());
+		assertNotNull(((List<SenderSyncArchive>) results.get("items")).get(0).getEventDate());
+		assertNotNull(((List<SenderSyncArchive>) results.get("items")).get(0).getDateSent());
+	}
+	
+	@Test
+	public void shouldDosearchByEventDateWithEndDate() {
+		
+		String stardDate = "";
+		String endDate = "2022-10-23";
+		
+		Map results = controller.doSearchByPeriod(stardDate, endDate, SenderSyncArchive.EVENT_DATE);
+		assertEquals(1, results.get("count"));
+		assertEquals(2, results.size());
+		assertEquals("e2bc25aa-1d5f-11e0-b929-000c29ad1d07",
```

Comment:
What is this assertion for? You should be checking that the item size matches expected and may be that the items ids or message uuids are match the expected